packages/adaptive_scaffold/example/lib/adaptive_layout_demo.dart

```diff
@@ -0,0 +1,102 @@
+  const _MyHomePage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    // Define the list of destinations to be used within the app.
+    const List<NavigationDestination> destinations = <NavigationDestination>[
+      NavigationDestination(
+          label: 'Inbox', icon: Icon(Icons.inbox, color: Colors.black)),
+      NavigationDestination(
+          label: 'Articles',
+          icon: Icon(Icons.article_outlined, color: Colors.black)),
+      NavigationDestination(
+          label: 'Chat',
+          icon: Icon(Icons.chat_bubble_outline, color: Colors.black)),
+      NavigationDestination(
+          label: 'Video',
+          icon: Icon(Icons.video_call_outlined, color: Colors.black)),
+    ];
+
+    // AdaptiveLayout has a number of slots that take SlotLayouts and these SlotLayouts' configs take maps of Breakpoints to SlotLayoutConfigs.
+    return AdaptiveLayout(
+      // Primary navigation config has nothing from 0 to 600, then an unextended
+      // NavigationRail then an extended NavigationRail.
+      primaryNavigation: SlotLayout(
+        config: <Breakpoint, SlotLayoutConfig?>{
+          Breakpoints.medium: SlotLayoutConfig(
+            inAnimation: AdaptiveScaffold.leftOutIn,
+            key: const Key('pnav1'),
+            builder: (_) =>
+                AdaptiveScaffold.toNavigationRail(destinations: destinations),
+          ),
+          Breakpoints.large: SlotLayoutConfig(
+            key: const Key('pnav2'),
+            inAnimation: AdaptiveScaffold.leftOutIn,
+            builder: (_) => AdaptiveScaffold.toNavigationRail(
+                extended: true, destinations: destinations),
+          ),
+        },
+      ),
+      // Body switches between a ListView and a GridView from small to medium breakpoints and onwards.
+      body: SlotLayout(
+        config: <Breakpoint, SlotLayoutConfig?>{
+          Breakpoints.small: SlotLayoutConfig(
+            key: const Key('body'),
+            builder: (_) => ListView.builder(
+              itemCount: 10,
+              itemBuilder: (_, __) => Padding(
+                padding: const EdgeInsets.all(8.0),
+                child: Container(
+                  color: const Color.fromARGB(255, 255, 201, 197),
+                  height: 400,
+                ),
+              ),
+            ),
+          ),
+          Breakpoints.mediumAndUp: SlotLayoutConfig(
+            key: const Key('body1'),
+            builder: (_) =>
+                GridView.count(crossAxisCount: 2, children: <Widget>[
+              for (int i = 0; i < 10; i++)
+                Padding(
+                  padding: const EdgeInsets.all(8.0),
+                  child: Container(
+                    color: const Color.fromARGB(255, 255, 201, 197),
+                    height: 400,
+                  ),
+                )
+            ]),
+          ),
+        },
+      ),
+      // BottomNavigation is only active in mobile.
```

Comment:
Does it happen if I make my window small on desktop?  Seems like it's not only mobile, it's "small" views.